### PR TITLE
linux-atm: fix format errors

### PR DESCRIPTION
**Pull request by neheb**
Could potentially fix crashing with big endian platforms.

### CODE REVIEW

**Comment by neheb** [comment]
Based on talks with Mr. Musl, he says crashing happens when a wrong format is used followed by %s. Eg.

```
"%d %s"
```

if a 64-bit value is passed to %d, the second half gets interpreted as part of the %s.

Weird that it mostly happens on big endian.

File: package/network/utils/linux-atm/patches/800-format.patch
Comment (robimarko) [inline package/network/utils/linux-atm/patches/800-format.patch:1]: Can you please make this patch applicable with git am and with a proper description.

**Comment by robimarko** [comment]
Looks sane with adding the length modifiers, I left a comment on the patch though.

**Comment by neheb** [comment]
Curious if this should be done honestly. Yes these are fixes, but I doubt there's any crashing that gets fixed.

File: package/network/utils/linux-atm/patches/800-format.patch [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+-			sprintf(outp,"%ld", t == 0 ? 0 : (r1->ru_ixrss-r0->ru_ixrss)/t);
++			sprintf(outp, PRIu64, t == 0 ? 0 : (uint64_t)(r1->ru_ixrss-r0->ru_ixrss)/t);
+ 			END(outp);
+ 			break;
+ 
+ 		case 'D':
+-			sprintf(outp,"%ld", t == 0 ? 0 :
+-			    (r1->ru_idrss+r1->ru_isrss-(r0->ru_idrss+r0->ru_isrss))/t);
++			sprintf(outp, PRIu64, t == 0 ? 0 :
++			    (uint64_t)(r1->ru_idrss+r1->ru_isrss-(r0->ru_idrss+r0->ru_isrss))/t);
```
Comment (hauke) [inline package/network/utils/linux-atm/patches/800-format.patch:135]: what type was used here before you added the cast? Adding the old error message to the comment should be fine.

File: package/network/utils/linux-atm/patches/800-format.patch
Comment (hauke) [reply package/network/utils/linux-atm/patches/800-format.patch:1]: @neheb

File: package/network/utils/linux-atm/patches/800-format.patch (hunk 1 above)
Comment (neheb) [reply package/network/utils/linux-atm/patches/800-format.patch:135]: long. Now it’s long long.

Comment (hauke) [reply package/network/utils/linux-atm/patches/800-format.patch:135]: if it was long then, the change is not needed.

Comment (neheb) [reply package/network/utils/linux-atm/patches/800-format.patch:135]: Sorry, under 32-bit it's long long. Under 64-bit it's long.

**Comment by neheb** [comment]
I'll close this. There's no crashing fixed here

**Pull request closed without merging**